src/python/pants/engine/internals/graph_test.py

```diff
@@ -409,35 +394,6 @@ def test_filesystem_specs_glob(self) -> None:
             ]
         )
 
-    def test_filesystem_specs_merge_when_same_address(self) -> None:
```

Comment:
It's impossible to trigger this edge case now. I didn't update the production code yet because it will be removed immediately after in a followup.